src/luascript.cpp

```diff
@@ -12475,6 +12481,59 @@ int LuaScriptInterface::luaPartySetSharedExperience(lua_State* L)
 	return 1;
 }
 
+int32_t LuaScriptInterface::luaDoCreateParty(lua_State* L)
+{
+	//doCreateParty(cid, invitedPlayer)
+	Player* player = getPlayer(L, 1);
+	if (player) {
+		Party* party = player->getParty();
+		if (!party) {
+			party = new Party(player);
+			Player* invitedPlayer = getPlayer(L, 2);
+			party->invitePlayer(*invitedPlayer);
+		} else if (party->getLeader() != player) {
+			std::ostringstream ss;
+			ss << player->getName() << " is already in a party.";
+			player->sendTextMessage(MESSAGE_INFO_DESCR, ss.str());
+			lua_pushboolean(L, false);
+		}
+	} else {
+		reportErrorFunc(getErrorDesc(LUA_ERROR_CREATURE_NOT_FOUND));
+		lua_pushboolean(L, false);
+	}
+	return 1;
+}
+
+int32_t LuaScriptInterface::luaDoJoinParty(lua_State* L)
+{
+	//doJoinParty(lider, invitedPlayer)
+	Player* player = getPlayer(L, 2);
+	if (!player) {
```

Comment:
If you're doing checking for errors/invalid inputs, please use if(somecondition) {report_errors; return 1} rather than if elses because it makes the code slightly harder to read.